### ISSUE TITLE
Support for Python 3.10

### ISSUE THREAD

**Issue by jgbos** [Kitware/SMQTK-Detection#40]
Is there any reason this repo cannot support Python 3.10?

**Comment by jgbos** [comment]
This package appears to be the only one one that does not support Python 3.10:

https://github.com/Kitware/SMQTK-Detection/blob/master/poetry.lock#L274

**Comment by Purg** [comment]
You're right! This is actually on my docket for this sprint to update this. The culprit is the numba dependency that I need to bifurcate the abstract requirement for. Currently, the requirement on `~0.53.0`, which effectively equates to `>=0.53.0,<0.54`, doesn't support beyond 3.9, likely due to the transitive dependency you pointed out. The solution I will be working towards is to specify a `>=` abstract requirement for python versions >=3.7. Given that xaitk-saliency has a lower bound of 3.7, it will strictly pick up the ">=" requirement, which should be interpreted appropriately to support all current python versions:
```toml
numba = [
    {version = "<0.54", optional = true, python = "<3.7" },
    {version = ">=0.54", optional = true, python = ">=3.7" },
]
```
The update for this repository would include testing on the new-currently-released python versions to make sure that there is an appropriately lockable version of numba and its dependencies for each python version greater than 3.9.

**Comment by Purg** [comment]
Attempted just updating numba (#41), but this has exposed that there are more issues related to 3.6 deprecation, mostly dealing with poetry and setuptools not playing nicely for 3.6. I think its more productive to proceed into the planned deprecation of supporting 3.6 than trying to find an intermediate work-around.

**Comment by Purg** [comment]
A more broad update is happening in #42.
